src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -546,14 +545,7 @@ public void Dispose()
                 }
                 finally
                 {
-                    lock (_throttleLock)
-                    {
-                        if (_throttle != null)
-                        {
-                            _throttle.Release();
-                            _throttle = null;
-                        }
-                    }
+                    Interlocked.Exchange(ref _throttle, null)?.Release();
```

Comment:
Correct, since this is a throttle on global HTTP I chose to be defensive in the releases. If two invocations of `Dispose` are executed at the same time, `_throttle` will release twice. This can be protected by an atomic exchange or a `lock`.